tools/gfx/slang.slang

```diff
@@ -0,0 +1,451 @@
+namespace slang
+{
+
+typedef int64_t Result;
+typedef uint64_t Size;
+typedef int64_t Int;
+typedef uint64_t UInt;
+
+/*!
+@brief Severity of a diagnostic generated by the compiler.
+Values come from the enum below, with higher values representing more severe
+conditions, and all values >= SLANG_SEVERITY_ERROR indicating compilation
+failure.
+*/
+enum SlangSeverity
```

Comment:
As long as kept in sync with slang.h, we could use names that are move `enum class`- like. 

```
enum SlangSeverity
{
    Disables = 0, 
    ///...
};
```